lewis_emulators/mecfrf/interfaces/stream_interface.py

```diff
@@ -0,0 +1,75 @@
+import six
+from lewis.adapters.stream import StreamInterface, Cmd
+import threading
+import struct
+
+
+EXPECTED_MESSAGE_LENGTH = 188
+
+
+class MecfrfStreamInterface(StreamInterface):
+
+    # Commands that we expect via serial during normal operation (No commands - the device always sends a stream of
+    # information to the IOC without being polled)
+    commands = {}
+
+    in_terminator = ""
+    out_terminator = ""
+
+    def __init__(self):
+        super(MecfrfStreamInterface, self).__init__()
+        self._queue_next_unsolicited_message()
+
+    def _queue_next_unsolicited_message(self):
+        timer = threading.Timer(1.0, self.get_data_unsolicited)
+        timer.daemon = True
+        timer.start()
+
+    def handle_error(self, request, error):
+        print("An error occurred at request " + repr(request) + ": " + repr(error))
+        return str(error)
+
+    def get_data_unsolicited(self):
+        self._queue_next_unsolicited_message()
+
+        if not self.device.connected:
+            return
+
+        try:
+            handler = self.handler
+        except AttributeError:
+            # Happens if no client is currently connected.
+            return
+        else:
+            if self.device.corrupted_messages:
+                # Nonsense message which should cause alarms in the IOC.
+                handler.unsolicited_reply(b"A" * EXPECTED_MESSAGE_LENGTH)
+            else:
+                handler.unsolicited_reply(self._construct_status_message())
```

Comment:
this is throwing 
```
Exception in thread Thread-129:
Traceback (most recent call last):
  File "C:\instrument\apps\python\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "C:\instrument\apps\python\lib\threading.py", line 1072, in run
    self.function(*self.args, **self.kwargs)
  File "C:\Instrument\Apps\EPICS\support\DeviceEmulator\master\lewis_emulators\mecfrf\interfaces\stream_interface.py", line 48, in get_data_unsolicited
    handler.unsolicited_reply(self._construct_status_message())
  File "C:\instrument\apps\python\lib\site-packages\lewis\adapters\stream.py", line 114, in unsolicited_reply
    self.push(b(reply + self._target.out_terminator))
  File "C:\instrument\apps\python\lib\asynchat.py", line 193, in push
    self.initiate_send()
  File "C:\instrument\apps\python\lib\asynchat.py", line 251, in initiate_send
    del self.producer_fifo[0]
IndexError: deque index out of range
```

Can you do anything about this or is it an issue in Lewis?

Reply:
Hmm, I didn't get this error on my machine. Did you have to do anything in particular to reproduce it? Is it consistent?